test/integration/nighthawk_test_server.py

```diff
@@ -18,61 +18,79 @@
 from test.integration.common import IpVersion, NighthawkException
 
 
+def _substitute_yaml_values(runfiles_instance, obj, params):
+  if isinstance(obj, dict):
+    for k, v in obj.items():
+      obj[k] = _substitute_yaml_values(runfiles_instance, v, params)
+  elif isinstance(obj, list):
+    for i in range(len(obj)):
+      obj[i] = _substitute_yaml_values(runfiles_instance, obj[i], params)
+  else:
+    if isinstance(obj, str):
+      # Inspect string values and substitute where applicable.
+      INJECT_RUNFILE_MARKER = '@inject-runfile:'
+      if obj[0] == '$':
+        return Template(obj).substitute(params)
+      elif obj.startswith(INJECT_RUNFILE_MARKER):
+        with open(runfiles_instance.Rlocation(obj[len(INJECT_RUNFILE_MARKER):].strip()),
+                  'r') as file:
+          return file.read()
+  return obj
+
+
 class TestServerBase(object):
   """
     Base class for running a server in a separate process.
+
+    Arguments:
```

Comment:
(nit) The "Args:" section of a class docstring actually belongs to the docstring of the initializer (__init__). Would you mind moving it there?

I am also happy to do this in a follow-up PR since this is minor, just let me know.

Reply:
I prefer to do it in a follow up: this file exists out of `benchmarks/`, I'm hoping to tighten up the format checking so that I can get all the python to comply with the check format rules that are now specific to `benchmark/`. Being backed by that makes it easier, as the doc string format isn't in my muscle memory yet. Added a note to https://github.com/envoyproxy/nighthawk/issues/371 as a reminder.

Reply:
Forgot to mark this is ready for review after posting the comment above  - just did so.